pkg/kubectl/cmd/proxy.go

```diff
@@ -145,6 +145,9 @@ func RunProxy(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
 	keepalive := cmdutil.GetFlagDuration(cmd, "keepalive")
 
 	server, err := proxy.NewServer(staticDir, apiProxyPrefix, staticPrefix, filter, clientConfig, keepalive)
+	if err != nil {
+		glog.Fatal(err)
```

Comment:
Don't `Fatalf` in CLI, just `return err`, the error will be properly handled one level higher.

Reply:
Can you also add a test case covering this case?

Reply:
A few lines below that an error is handled in a Fatal way. Looks like this function must always return Nil.
About testing, how to simulate and check for fatal?

Reply:
The other Fatal is also wrong, the method should return err when fails.